### PR TITLE
populate machine/node status with IPv6 address

### PR DESCRIPTION
**Pull request by PratikDeoghare**
**What this PR does / why we need it**:
This populates stats of Machine CRD with IPv6 address information. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>` format, will close the issue(s) when PR gets merged)*:
Story https://github.com/kubermatic/kubermatic/issues/9294

**Optional Release Note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/cloudprovider/provider/aws/provider.go [hunk 1]
```diff
@@ -799,6 +799,8 @@ func (p *provider) Create(machine *clusterv1alpha1.Machine, data *cloudprovidert
 
 	if util.ContainsCIDR(networkConfig.PodCIDRs, util.IPv6) {
 		instanceRequest.NetworkInterfaces[0].Ipv6AddressCount = aws.Int64(1)
+	} else {
+		klog.Infoln("PodCIDRs don't contain IPv6")
```
Comment (moadqassem) [inline pkg/cloudprovider/provider/aws/provider.go:803]: nit:
```suggestion
		klog.Infoln("no IPv6 found in for PodCIRD in network configs")
```

File: pkg/cloudprovider/provider/aws/provider.go [hunk 2]
```diff
@@ -1007,6 +1009,15 @@ func (d *awsInstance) Addresses() map[string]v1.NodeAddressType {
 		aws.StringValue(d.instance.PrivateDnsName):   v1.NodeInternalDNS,
 	}
 
+	for _, netInterface := range d.instance.NetworkInterfaces {
+		for _, addr := range netInterface.Ipv6Addresses {
+			ipAddr := aws.StringValue(addr.Ipv6Address)
```
Comment (moadqassem) [inline pkg/cloudprovider/provider/aws/provider.go:1014]: Wouldn't this work without the casting? 
```suggestion
			ipAddr := addr.String()
```

File: pkg/cloudprovider/util/net.go [hunk 3]
```diff
@@ -65,3 +65,9 @@ func ContainsCIDR(cidrs []string, version IPVersion) bool {
 
 	return false
 }
+
+// IsLinkLocal checks if given ip address is link local
```
Comment (moadqassem) [inline pkg/cloudprovider/util/net.go:69]: Can you please explain more, why do we need only locally linked ips?

Comment (PratikDeoghare) [reply pkg/cloudprovider/util/net.go:69]: This is used to filter them out because showing them in the machine status is not useful.

Comment (PratikDeoghare) [reply pkg/cloudprovider/util/net.go:69]: ah! understood. :) 
Will add in next PRs. All the e2e tests are green don't want to babysit them right now.

**Comment by PratikDeoghare** [comment]
Covered in https://github.com/kubermatic/machine-controller/pull/1242

**Pull request closed without merging**